hangman.py

```diff
@@ -0,0 +1,338 @@
+    if len(my_word) != len(other_word):
+        return False
+    for symbol in range(0, len(my_word)):
+        if my_word[symbol] == '_' or my_word[symbol] == other_word[symbol]:
+            continue
+        return False
+   #Code below is needed for not showing words, if they have same letters and length,
+   #but my_word has any letter and other_word has two same letters as letter in my_word
+   #For example, function doesn't show word 'apple' if my_word is 'a_ ple'
+    for symbol in range(0, len(my_word_for_dublicate)):
+        #Counting number of every letter in both words and comparing them
+        letter_occurrence_my_word = my_word_for_dublicate.count(my_word_for_dublicate[symbol])
+        letter_occurrence_other_word = other_word.count(my_word_for_dublicate[symbol])
+        if letter_occurrence_my_word != letter_occurrence_other_word:
+            return False
+
+    return True
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+
+    '''
+    hint_list = []
+    my_word = my_word.replace('_ ', '_')
+    my_word_for_dublicate = my_word.replace('_', '')
+    for other_word in wordlist:
+        if match_with_gaps(my_word, other_word, my_word_for_dublicate):
+            hint_list.append(other_word)
+    if hint_list == []:
+        print("No matches found")
+    else:
+        print("Possible word matches are:", ' '.join(hint_list))
+
+
+def hangman_with_hints(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Make sure to check that the user guesses a letter
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    * If the guess is the symbol *, print out all words in wordlist that
+      matches the current guessed word.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    hangman(secret_word, hints_switch = True)
+
+
+if __name__ == "__main__":
+    secret_word = choose_word(wordlist)
```

Comment:
This variable is used as a global in some places

Reply:
done